snapshot/snapshotfs/upload.go

```diff
@@ -1192,11 +1193,13 @@ func (u *Uploader) maybeOpenDirectoryFromManifest(ctx context.Context, man *snap
 
 // Upload uploads contents of the specified filesystem entry (file or directory) to the repository and returns snapshot.Manifest with statistics.
 // Old snapshot manifest, when provided can be used to speed up uploads by utilizing hash cache.
+// labels is applied to every checkpoint of a backup.
 func (u *Uploader) Upload(
 	ctx context.Context,
 	source fs.Entry,
 	policyTree *policy.Tree,
 	sourceInfo snapshot.SourceInfo,
+	labels map[string]string,
```

Comment:
this looks like an optional parameter with `nil` being a reasonable default, can we add a field to `Uploader` instead?